msu/systems/serialization/serialization_types/array_serialization_data.nut

```diff
@@ -0,0 +1,55 @@
+::MSU.Class.ArrayData <- class extends ::MSU.Class.AbstractData
+{
+	__DataArray = null;
+
+	constructor( _data )
```

Comment:
In your example you pass an empty array here - just have a default null and initialize it here?

Reply:
Sorry the example was not updated. It should be instantiating `::MSU.Class.SerializationData()` and that can definitely be initialized without passing any argument as it defaults to null and instantiates the array if null. For the `ArrayData` and `TableData` I require passing an array/table because they are meant to convert the given array/table to the data and do not have the possibility to later add data to it. The `SerializationData` class is the one that can take data after it has been instantiated.